### PR TITLE
fix: crash on screenshot sharing

### LINKED ISSUE #76: Share Extension Crashes when sharing a screenshot
When I try to share a screenshot on iPhone, the extension crashes as soon as a I call `ShareMenuReactView.data()`.  Works fine if I save the screenshot to photos and then share the photo from the photos app.  Is there a different way of handling screenshot shares, or is this a bug?

### PR DESCRIPTION
**Pull request by dabakovich**
When we sharing a screenshot the share extension receives `UIImage` object not URL. So I handled the case.

Fixes #76.

### CODE REVIEW

**Comment by dabakovich** [comment]
Do you have any comments here guys?
@caiosba, @Gustash

**Comment by caiosba** [comment]
It looks good to me, but @Gustash is the iOS maintainer, so I defer to him. Thanks for your contribution, btw!

**Comment by Rob117** [comment]
@dabakovich 

I ran into something interesting trying to use this commit. So, it seems that when you try to share a screenshot, the file location is correctly passed to react native. However, when `continueInApp` is called in JS the actual app is no longer opened. This is not the case with other forms of data such as text strings or local files (photo reel, files). I suspect it might have something to do with the temporary nature of the screenshot (/tmp), but I'm not sure. Did you see this when you were testing?

Edit:
The error appears to be happening at the native level, getting a EXC_BAD_INSTRUCTION from the sharer extension.

**Comment by Rob117** [comment]
Further investigation:
Something about image conversion, mabye?
com.apple.CoreSimulator.SimDevice.EB16F52C-0E45-406B-B7B5-A9E13387F69D[55628] (com.apple.xpc.launchd.domain.pid.ShareExtension.48629): Could not get domain target from bundle: path = /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/PrivateFrameworks/MediaConversionService.framework/XPCServices/com.apple.photos.ImageConversionService.xpc, error = 22: Invalid argument

**Comment by Rob117** [comment]
Okay, I found the problem, and an _ugly_ solution. Please refactor this so it fits btter - I have no idea what I'm doing with swift, but I managed to debug the issue.

When an image is sent over the wire as a screenshot, the `ShareViewController.swift`:`storeFile` function is getting the UIIMage, not the ImageUrl that you write the screenshot to. I don't know why - I did manage to get a super hacky solution by simply checking to see if the passed over data pointed to an image, and if so, writing that data to a file:

Replace the method storeFile in ShareViewController.swift with this:
```swift
func storeFile(withProvider provider: NSItemProvider) {
    print("Storing File");
    print("Provider: \(provider)")
    provider.loadItem(forTypeIdentifier: kUTTypeData as String, options: nil) { (data, error) in
      guard (error == nil) else {
        self.exit(withError: error.debugDescription)
        return
      }
      var finalURL:URL! = nil;
      let imgData: UIImage! = data as? UIImage;
      if (imgData != nil) {
          let imageData: Data! = imgData.pngData();

          // Creating temporary URL for image data (UIImage)
          guard let imageURL = NSURL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent("TemporaryScreenshot.png") else {
            print("Error getting image url file location to write to");
            self.exit(withError: COULD_NOT_FIND_IMG_ERROR)
            return;
          }
        do {
          // Writing the image to the URL
          try imageData.write(to: imageURL)
          finalURL = imageURL;
        } catch {
          print("Error writing image data");
          self.exit(withError: COULD_NOT_FIND_IMG_ERROR)
        }
      } else {
        guard let url = data as? URL else {
          print("Error turning data into url");
          self.exit(withError: COULD_NOT_FIND_IMG_ERROR)
          return
        }
        finalURL = url;
      }
      
      guard let hostAppId = self.hostAppId else {
        self.exit(withError: NO_INFO_PLIST_INDENTIFIER_ERROR)
        return
      }
      guard let userDefaults = UserDefaults(suiteName: "group.\(hostAppId)") else {
        self.exit(withError: NO_APP_GROUP_ERROR)
        return
      }
      guard let groupFileManagerContainer = FileManager.default
              .containerURL(forSecurityApplicationGroupIdentifier: "group.\(hostAppId)")
      else {
        self.exit(withError: NO_APP_GROUP_ERROR)
        return
      }
      
      let mimeType = finalURL.extractMimeType()
      let fileExtension = finalURL.pathExtension
      let fileName = UUID().uuidString
      let filePath = groupFileManagerContainer
        .appendingPathComponent("\(fileName).\(fileExtension)")
      
      guard self.moveFileToDisk(from: finalURL, to: filePath) else {
        self.exit(withError: COULD_NOT_SAVE_FILE_ERROR)
        return
      }
      
      userDefaults.set([DATA_KEY: filePath.absoluteString,  MIME_TYPE_KEY: mimeType],
                       forKey: USER_DEFAULTS_KEY)
      userDefaults.synchronize()
      
      self.openHostApp()
    }
  }
```

**Comment by nromptea42** [comment]
Hi, is there any plan of merging this ?

**Comment by yungbigos** [comment]
@caiosba Hi there! 😊 This would be a super useful addition. Is there anything blocking the merge?

**Review by caiosba — APPROVED** [review]

**Comment by caiosba** [comment]
Thanks!

**Pull request merged**
